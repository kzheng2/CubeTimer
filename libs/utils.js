const pad = n => n < 10 ? `0${n}` : n

export const getTimeObj = (lapse) => {
  const millisecond = pad(Math.floor((lapse % 1000) / 10))
  const second = pad(Math.floor(lapse / 1000))
  const minute = pad(Math.floor(second / 60))
  return { minute, second, millisecond }
}

export const isSpace = (event) => event.which === 32
