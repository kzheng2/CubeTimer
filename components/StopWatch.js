import React from 'react'
import { inject, observer } from 'mobx-react'
import { getTimeObj } from '../libs/utils'

@inject('store')
@observer
class StopWatch extends React.Component {
  render () {
    const { lapse, running, standby, isServer } = this.props.store
    const { minute, second, millisecond } = getTimeObj(lapse)
    const value = isServer ? '...' : `${minute}:${second}:${millisecond}`
    return (
      <div style={{
        boxShadow: 'rgba(0, 0, 0, 0.2) 0px 2px 4px 0px, rgba(0, 0, 0, 0.1) 0px 25px 50px 0px'
      }}>
        <div style={{
          background: '#FFF',
          color: (running || standby) ? 'red' : 'black',
          font: '3rem menlo, monaco, monospace',
          width: '5.5em',
          padding: '1rem',
          userSelect: 'none',
          WebkitUserSelect: 'none',
          textAlign: 'center'
        }}>
          {value}
        </div>
      </div>
    )
  }
}

export default StopWatch
