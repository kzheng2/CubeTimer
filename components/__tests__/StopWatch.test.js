/* global it, expect, describe */
import React from 'react'
import StopWatch from '../StopWatch'
import { getTimeObj } from '../../libs/utils'
import { shallow } from 'enzyme'

describe('StopWatch', () => {
  it('render without crash', () => {
    const el = shallow(<StopWatch store={{}} />)
    expect(el).toBeTruthy()
  })

  it('render running status', () => {
    const store = { running: true }
    const getTarget = el => el.find('div > div')
    let el = shallow(<StopWatch store={store} />)

    expect(getTarget(el).is({ style: { color: 'red' } })).toBe(true)

    store.running = false
    el = shallow(<StopWatch store={store} />)

    expect(getTarget(el).is({ style: { color: 'black' } })).toBe(true)
  })

  it('render correct time', () => {
    const store = { lapse: Math.random() }
    const el = shallow(<StopWatch store={store} />)
    const { minute, second, millisecond } = getTimeObj(store.lapse)

    expect(el.text()).toBe(`${minute}:${second}:${millisecond}`)
  })
})
