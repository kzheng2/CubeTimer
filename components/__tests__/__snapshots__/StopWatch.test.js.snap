// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopWatch render correctly 1`] = `
<div
  className="timebar stopwatch"
  style={
    Object {
      "color": "black",
    }
  }
>
  00:00:04
</div>
`;

exports[`StopWatch render correctly 2`] = `
<div
  className="timebar stopwatch"
  style={
    Object {
      "color": "red",
    }
  }
>
  00:00:04
</div>
`;

exports[`StopWatch render correctly 3`] = `
<div
  className="timebar stopwatch"
  style={
    Object {
      "color": "black",
    }
  }
>
  00:00:00
</div>
`;
